src/components/ErrorBoundary.tsx

```diff
@@ -0,0 +1,51 @@
+import { Component, type ReactNode } from 'react';
+import '../styles/ErrorBoundary.scss';
+
+interface Props {
+  children: ReactNode;
+}
+
+interface State {
+  message: string | null;
+}
+
+class ErrorBoundary extends Component<Props, State> {
+  timeoutId: number | null = null;
+
+  constructor(props: Props) {
+    super(props);
+    this.state = { message: null };
+  }
+
+  componentDidMount() {
+    window.addEventListener('GlobalError', this.handleGlobalError);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('GlobalError', this.handleGlobalError);
+    if (this.timeoutId) clearTimeout(this.timeoutId);
+  }
+
+  handleGlobalError = (e: Event) => {
+    const customEvent = e as CustomEvent;
+    const message = customEvent.detail?.message || 'Unknown error occurred';
+    this.setState({ message });
+
+    this.timeoutId = window.setTimeout(() => {
+      this.setState({ message: null });
+    }, 5000);
+  };
+
+  render() {
+    const { message } = this.state;
+
+    return (
+      <>
+        {message && <div className="toast">{message}</div>}
+        {this.props.children}
+      </>
+    );
+  }
+}
```

Comment:
then there will be something like this

```suggestion
render() {
    if (this.state.error) {
      return (
        <div>
          <h2>Something went wrong</h2>
          <p>{this.state.message}</p>
          <button onClick={() => this.setState({ message: null, error: null })}>
            Reset
          </button>
        </div>
      );
    }
    return this.props.children;
  }
```